src/parser/core/modules/Statuses.js

```diff
@@ -0,0 +1,180 @@
+import _ from 'lodash'
+
+import ACTIONS from 'data/ACTIONS'
+import STATUSES, {STATUS_EFFECT_TYPES} from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {ItemGroup, Item} from './Timeline'
+import {getDataBy} from 'data'
+
+// Track the cooldowns on actions and shit
+export default class Statuses extends Module {
+	static handle = 'statuses'
+	static dependencies = [
+		'timeline',
+		// need this module to be executed after gcd to have gcd group created in timeline
+		'cooldowns', // eslint-disable-line @xivanalysis/no-unused-dependencies
+		'gcd', // eslint-disable-line @xivanalysis/no-unused-dependencies
+	]
+
+	STATUSES_STACK_MAPPING = {
+		[STATUSES.WALKING_DEAD.id]: STATUSES.LIVING_DEAD.id,
+		[STATUSES.GIANT_DOMINANCE.id]: STATUSES.EARTHLY_DOMINANCE.id,
+		[STATUSES.DIVINE_VEIL_AFTER_HEAL.id]: STATUSES.DIVINE_VEIL.id,
+	}
+
+	_statuses = [];
+	_groups = {}
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('complete', this._onComplete)
+		this.addHook('applybuff', {by: 'player'}, this._onApply)
+		this.addHook('applydebuff', {by: 'player'}, this._onApply)
+		this.addHook('refreshdebuff', {by: 'player'}, this._onRefresh)
+		this.addHook('refreshbuff', {by: 'player'}, this._onRefresh)
+		this.addHook('removebuff', {by: 'player'}, this._onRemove)
+		this.addHook('removedebuff', {by: 'player'}, this._onRemove)
+
+		this.addHook('applybuff', {by: 'pet'}, this._onApply)
+		this.addHook('applydebuff', {by: 'pet'}, this._onApply)
+		this.addHook('refreshdebuff', {by: 'pet'}, this._onRefresh)
+		this.addHook('refreshbuff', {by: 'pet'}, this._onRefresh)
+		this.addHook('removebuff', {by: 'pet'}, this._onRemove)
+		this.addHook('removedebuff', {by: 'pet'}, this._onRemove)
+	}
+
+	_onApply(event) {
+		if (this.parser.report.friendlyPets.some(p => p.id === event.targetID)) {
+			return
+		}
+		const status = getDataBy(STATUSES, 'id', event.ability.guid)
+
+		if (!status) {
+			return
+		}
+
+		this._addStatus(event, status)
+	}
+
+	_onRefresh(event) {
+		if (this.parser.report.friendlyPets.some(p => p.id === event.targetID)) {
+			return
+		}
+
+		const status = getDataBy(STATUSES, 'id', event.ability.guid)
+
+		if (!status) {
+			return
+		}
+
+		this._updatePrevStatus(event, status)
+
+		this._addStatus(event, status)
+	}
+
+	_onRemove(event) {
+		if (this.parser.report.friendlyPets.some(p => p.id === event.targetID)) {
+			return
+		}
+
+		const status = getDataBy(STATUSES, 'id', event.ability.guid)
+
+		if (!status) {
+			return
+		}
+
+		this._updatePrevStatus(event, status)
+	}
+
+	_updatePrevStatus(event, status) {
+		const prev = _.findLast(this._statuses, it => it.status.id === status.id)
+
+		if (prev) {
+			prev.end = event.timestamp - this.parser.fight.start_time
+		}
+	}
+
+	_addStatus(event, status) {
+		if (this._statuses.some(it => {
+			const diff = Math.abs(event.timestamp - this.parser.fight.start_time - it.start)
+			return it.status.id === status.id && diff <= 2 * 1000
```

Comment:
`2 * 1000` should be best represented as a constant somewhere, so that it's obvious what this value means.